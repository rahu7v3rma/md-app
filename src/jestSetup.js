import mockRNCameraRoll from '@react-native-camera-roll/camera-roll/src/__mocks__/nativeInterface';
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock';
import mockRNDeviceInfo from 'react-native-device-info/jest/react-native-device-info-mock';

import { MockData } from './utils/mockData';

// setImmediate and clearImmediate aren't mocked by useFakeTimers if they are
// not set to a function (https://github.com/jestjs/jest/pull/11599)
global.setImmediate = jest.fn();
global.clearImmediate = jest.fn();
jest.useRealTimers();

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

jest.mock('react-native-keychain', () => {
    return {
        SECURITY_LEVEL_ANY: 'MOCK_SECURITY_LEVEL_ANY',
        SECURITY_LEVEL_SECURE_SOFTWARE: 'MOCK_SECURITY_LEVEL_SECURE_SOFTWARE',
        SECURITY_LEVEL_SECURE_HARDWARE: 'MOCK_SECURITY_LEVEL_SECURE_HARDWARE',
        setGenericPassword: jest.fn().mockResolvedValue(),
        getGenericPassword: jest.fn().mockResolvedValue(false),
        resetGenericPassword: jest.fn().mockResolvedValue()
    };
});

jest.mock('react-native-splash-screen', () => {
    return {
        hide: jest.fn()
    };
});

export let mockUnwrap = jest.fn().mockImplementation(() => {
    return Promise.resolve({});
});
export const mockDispatch = jest.fn().mockImplementation(() => {
    return {
        unwrap: mockUnwrap
    };
});
jest.mock('@/hooks', () => ({
    useAppDispatch: () => mockDispatch,
    useChannelPreviewInfo: () => mockDispatch
}));
export const mockChannelId = 'mockChannelId';
export const mockChatClient = jest.fn().mockImplementation(() => ({
    userID: 1,
    chat_id: 1,
    queryChannels: jest.fn().mockImplementation(() => {
        return Promise.resolve([
            {
                cid: 99
            }
        ]);
    }),
    activeChannels: {
        mockChannelId
    }
}));
jest.mock('@/contexts/appChat', () => ({
    useAppChat: () => {
        return {
            chatClient: mockChatClient(),
            setActiveChatChannel: jest.fn()
        };
    }
}));
export const mockRNSha256 = jest
    .fn()
    .mockImplementation(() => Promise.resolve('123'));
jest.mock('react-native-sha256', () => {
    return {
        sha256: mockRNSha256
    };
});

jest.mock('react-native-share', () => ({
    default: jest.fn()
}));

jest.mock('@/services/api', () => ({
    getImageUploadCredentials: jest.fn(() => ({
        keyPrefix: 'keyPrefix',
        fileName: 'fileName',
        bucketName: 'bucketName'
    }))
}));

jest.mock('@/reducers/content', () => ({
    ContentSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        journey: MockData.journey,
        navigationKey: 'Main.Home',
        activeBlock: MockData.activeBlock,
        activeLesson: {
            lesson: {}
        },
        journeyBlocks: MockData.journeyBlocks
    })),
    loadJourney: jest.fn(),
    getActiveLesson: jest.fn()
}));

export const mockChatProfile = jest.fn(() => ({
    apiKey: 'apiKey',
    token: 'token',
    userId: 'userId',
    firebasePushProviderName: 'firebasePushProviderName'
}));
export const mockDeleteAccount = jest.fn();
jest.mock('@/reducers/user', () => ({
    UserSelectors: jest.fn().mockImplementation(() => ({
        loading: false,
        userInfo: {
            username: 'user1',
            first_name: 'Some',
            last_name: 'Name',
            email: 'user1@example.example'
        },
        userProfile: {
            diabetes_type: 'Prediabetes',
            image: null,
            onboarding_form_url: null
        },
        coach: {
            first_name: 'name',
            profile_image: 'profile_image'
        },
        chatProfile: mockChatProfile(),
        hasCoachChat: false
    })),
    userLogin: jest.fn(),
    changePassword: jest.fn(),
    resetPassword: jest.fn(),
    resetPasswordConfirm: jest.fn(),
    resetPasswordVerify: jest.fn(),
    getCoach: jest.fn(),
    refreshProfileSession: jest.fn(),
    getProfile: jest.fn(),
    searchFilterAction: jest.fn(),
    clearSearchText: jest.fn(),
    deleteAccount: mockDeleteAccount
}));
export const mockRNFSExists = jest
    .fn()
    .mockImplementation(() => Promise.resolve(false));
export const mockRNFSCachesDirectoryPath = 'mockCachesDirectoryPath';
export const mockRNFSDownloadFile = jest.fn().mockImplementation(() => ({
    promise: Promise.resolve({ statusCode: 200 })
}));
jest.mock('react-native-fs', () => {
    return {
        readFile: jest.fn(),
        exists: mockRNFSExists,
        CachesDirectoryPath: mockRNFSCachesDirectoryPath,
        downloadFile: mockRNFSDownloadFile,
        unlink: jest.fn().mockImplementation(() => Promise.resolve())
    };
});
jest.mock('react-native-image-crop-picker', () => {
    return {
        openPicker: jest.fn(),
        launchImageLibrary: jest.fn(),
        openCamera: jest.fn()
    };
});

jest.mock('@react-native-camera-roll/camera-roll', () => mockRNCameraRoll);
jest.mock('react-native-reanimated', () => ({
    ...require('react-native-reanimated/mock')
}));
jest.mock('react-native-device-info', () => mockRNDeviceInfo);
jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');

jest.mock('@/reducers/track', () => ({
    TrackSelectors: jest.fn().mockImplementation(() => ({
        trackFastState: '',
        trackStartedAt: '',
        trackTimeLimitInSeconds: 57600,
        trackedTimeInSeconds: 0
    })),
    trackStart: jest.fn(),
    trackUpdate: jest.fn()
}));

jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo);

jest.mock('@notifee/react-native', () =>
    require('@notifee/react-native/jest-mock')
);

export const mockOnTokenRefresh = jest.fn();
jest.mock('@react-native-firebase/messaging', () => ({
    __esModule: true,
    default: () => ({ onTokenRefresh: mockOnTokenRefresh }),
    FirebaseMessagingTypes: jest.fn()
}));

export const mockNavigate = jest.fn();
export const mockPop = jest.fn();
export const mockGoBack = jest.fn();
export const mockReplace = jest.fn();
export const mockNavigationDispatch = jest.fn();
export const mockUseRoute = jest.fn().mockImplementation(() => ({
    params: {
        id: 1,
        unit: 'kg',
        code: '11111',
        url: 'https://www.google.com/'
    }
}));

jest.mock('@/services/notification', () => ({
    resetNotifications: jest.fn(),
    registerBackgroundNotificationsListener: jest.fn(),
    getFCMToken: jest.fn().mockImplementation(() => {
        return Promise.resolve({});
    }),
    triggerFirebaseForegroundNotification: jest.fn(),
    registerFirebaseMessageListener: jest.fn(),
    requestNotificationPermissions: jest.fn()
}));

jest.mock('@react-navigation/native', () => {
    const real = jest.requireActual('@react-navigation/native');
    return {
        ...real,
        useNavigation: jest.fn().mockImplementation(() => {
            return {
                canGoBack: jest.fn().mockImplementation(() => {
                    return true;
                }),
                goBack: mockGoBack,
                navigate: mockNavigate,
                pop: mockPop,
                replace: mockReplace,
                dispatch: mockNavigationDispatch
            };
        }),
        useRoute: mockUseRoute,
        useIsFocused: jest.fn().mockImplementation(() => ({
            isFocused: true
        })),
        useFocusEffect: jest.fn()
    };
});

export const mockToast = jest.fn();

jest.mock('react-native-toast-message', () => ({
    show: mockToast
}));

export const mockGetChannelDisplayImage = jest.fn();
jest.mock('@/services/chat', () => ({
    getChannelDisplayImage: mockGetChannelDisplayImage
}));

export const mockConnectUser = jest.fn(() => Promise.resolve());
export const mockDisconnectUser = jest.fn();
export const mockGetInstance = jest.fn(() => ({
    connectUser: mockConnectUser,
    disconnectUser: mockDisconnectUser
}));
jest.mock('stream-chat', () => ({
    StreamChat: {
        getInstance: mockGetInstance
    }
}));

jest.mock('@aws-sdk/client-s3', () => ({
    S3Client: jest.fn(() => ({
        send: jest.fn(() => Promise.resolve())
    })),
    PutObjectCommand: jest.fn().mockImplementation(() => {
        return {};
    })
}));

jest.mock('react-native-pdf', () => null);
