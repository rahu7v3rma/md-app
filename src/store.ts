import AsyncStorage from '@react-native-async-storage/async-storage';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import {
    createMigrate,
    FLUSH,
    PAUSE,
    PERSIST,
    PersistedState,
    persistReducer,
    persistStore,
    PURGE,
    REGISTER,
    REHYDRATE
} from 'redux-persist';

import clientConfigReducer from '@/reducers/clientConfig';
import contentReducer from '@/reducers/content';
import logReducer from '@/reducers/log';
import notificationReducer from '@/reducers/notification';
import trackReducer from '@/reducers/track';
import userReducer from '@/reducers/user';

const rootReducer = combineReducers({
    clientConfig: clientConfigReducer,
    content: contentReducer,
    user: userReducer,
    log: logReducer,
    track: trackReducer,
    notification: notificationReducer
});

const migrations = {
    2: (state: PersistedState): PersistedState => {
        // add initial pickerValues - convert to `any` because of an issue with
        // redux-persist's typings -
        // https://github.com/rt2zz/redux-persist/issues/1065
        (state as any).log.pickerValues = {
            hydration: { units: [] },
            weight: { units: [] },
            glucose: { measurement_types: [], units: [] },
            medication: { drugs: [], doses: [] },
            insulin: { injection_types: [] },
            exercise: { types: [], intensities: [] }
        };

        return state;
    }
};

const persistConfig = {
    key: 'root',
    version: 2,
    storage: AsyncStorage,
    blacklist: ['clientConfig'],
    migrate: createMigrate(migrations, { debug: false })
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            // disable the serializable check for redux-persist's actions
            serializableCheck: {
                ignoredActions: [
                    FLUSH,
                    PAUSE,
                    PERSIST,
                    PURGE,
                    REGISTER,
                    REHYDRATE
                ]
            }
        })
});

export default store;

export const persistor = persistStore(store);

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

// reset persisted store when a user logs out
export const resetStore = () => {
    persistor.purge().then(() => persistor.flush());
};
