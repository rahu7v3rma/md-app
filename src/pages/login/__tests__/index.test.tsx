import React from 'react';
import 'react-native';
import Toast from 'react-native-toast-message';
import { Provider } from 'react-redux';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';

import { mockNavigate } from '@/jestSetup';
import LogIn from '@/pages/login';
import store from '@/store';

let mockUnwrap: any;
jest.mock('@/hooks', () => ({
    useAppDispatch: () =>
        jest.fn().mockImplementation(() => ({
            unwrap: mockUnwrap
        }))
}));
let tree: ReactTestRenderer;

describe('Login', () => {
    beforeAll(async () => {
        tree = renderer.create(
            <Provider store={store}>
                <LogIn />
            </Provider>
        );
        const emailInput = tree.root.findByProps({
            testID: 'emailInput'
        }).props;
        const passwordInput = tree.root.findByProps({
            testID: 'passwordInput'
        }).props;
        await act(() => emailInput.onChangeText('test@gmail.com'));
        await act(() => passwordInput.onChangeText('12345678'));
    });

    it('login Snapshot Testing with Jest', async () => {
        expect(tree).toMatchSnapshot();
    });

    it('login renders correctly', async () => {
        expect(tree.root).toBeTruthy();
    });

    it('Login toggles password visibility', async () => {
        const passwordInputIcon = tree.root.findByProps({
            testID: 'passwordInputIcon'
        }).props;
        act(() => {
            passwordInputIcon.onPress();
        });
        const passwordInput = tree.root.findByProps({
            testID: 'passwordInput'
        }).props;
        expect(passwordInput.secureTextEntry).toBe(false);
    });

    it('login calls userLogin action and navigates to Main on successful login', async () => {
        mockUnwrap = jest.fn().mockResolvedValue({});
        const signInButton = tree.root.findByProps({ testID: 'signInButton' });
        await act(() => signInButton.props.onPress());
        expect(mockNavigate).toBeCalledWith('Main');
    });

    it('login navigates to ChangePassword on 403 status code', async () => {
        mockUnwrap = jest.fn().mockRejectedValue({ status: 403 });
        const signInButton = tree.root.findByProps({ testID: 'signInButton' });
        await act(() => signInButton.props.onPress());
        expect(mockNavigate).toBeCalledWith('ChangePassword', {
            email: 'test@gmail.com'
        });
    });

    it('login displays error message for 401 status code', async () => {
        mockUnwrap = jest.fn().mockRejectedValue({ status: 401 });
        const signInButton = tree.root.findByProps({ testID: 'signInButton' });
        await act(() => signInButton.props.onPress());
        expect(Toast.show).toBeCalledWith({
            type: 'errorResponse',
            text1: 'Incorrect username or password',
            position: 'bottom'
        });
    });

    it('login displays error message for unknown error', async () => {
        mockUnwrap = jest.fn().mockRejectedValue({ status: 402 });
        const signInButton = tree.root.findByProps({ testID: 'signInButton' });
        await act(() => signInButton.props.onPress());
        expect(Toast.show).toBeCalledWith({
            type: 'errorResponse',
            text1: 'An unknown error has occurred',
            position: 'bottom'
        });
    });
});
