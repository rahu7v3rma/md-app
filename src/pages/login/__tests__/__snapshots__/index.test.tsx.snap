// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login login Snapshot Testing with Jest 1`] = `
<RCTSafeAreaView
  accessible={false}
  emulateUnlessSupported={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "height": 1334,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "flex-start",
            "paddingHorizontal": 20,
            "paddingTop": 30,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={96}
          bbWidth={96}
          fill="none"
          focusable={false}
          height={96}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 96,
                "width": 96,
              },
            ]
          }
          vbHeight={96}
          vbWidth={96}
          width={96}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGRect
              fill={4294967295}
              height={96}
              propList={
                Array [
                  "fill",
                ]
              }
              rx={32}
              width={96}
              x={0}
              y={0}
            />
            <RNSVGPath
              d="M37.645 65H26l2.874-33H39.34l8.623 16.279L56.218 32h10.686L70 65H58.429l-1.032-13.775L50.395 65h-4.938L38.53 51.225 37.645 65z"
              fill={4285700459}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          adjustsFontSizeToFit={true}
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
              },
              Object {
                "color": "#4D4F4D",
                "marginVertical": 32,
                "textAlign": "center",
              },
              Object {
                "color": "#4D4F4D",
                "fontSize": 28,
                "fontWeight": "700",
                "lineHeight": 36,
              },
            ]
          }
        >
          Welcome to
Mastering Programs
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 61,
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "shadowColor": "rgba(0, 0, 0, 0.3)",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0.5,
                "shadowRadius": 1,
              },
              Object {},
              false,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingHorizontal": 10,
                  "position": "relative",
                },
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              autoFocus={false}
              editable={true}
              keyboardType="email-address"
              multiline={false}
              numberOfLines={1}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Email*"
              placeholderTextColor="#4D4F4D"
              returnKeyType="done"
              secureTextEntry={false}
              selectionColor="#72996B"
              style={
                Array [
                  Object {
                    "color": "#4D4F4D",
                    "flex": 1,
                    "fontFamily": "Poppins",
                    "fontSize": 16,
                    "fontWeight": "500",
                  },
                  Object {},
                  Object {
                    "color": "#4D4F4D",
                    "fontFamily": "Poppins",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "lineHeight": 22,
                  },
                ]
              }
              testID="emailInput"
              value="test@gmail.com"
            />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginRight": 5,
                "opacity": 1,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "transparent",
                "borderRadius": 20,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 61,
                "justifyContent": "space-between",
                "paddingHorizontal": 10,
                "shadowColor": "rgba(0, 0, 0, 0.3)",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0.5,
                "shadowRadius": 1,
              },
              Object {},
              false,
              Object {
                "marginTop": 12,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "paddingHorizontal": 10,
                  "position": "relative",
                },
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={true}
              autoFocus={false}
              editable={true}
              keyboardType="default"
              multiline={false}
              numberOfLines={1}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Password*"
              placeholderTextColor="#4D4F4D"
              returnKeyType="done"
              secureTextEntry={true}
              selectionColor="#72996B"
              style={
                Array [
                  Object {
                    "color": "#4D4F4D",
                    "flex": 1,
                    "fontFamily": "Poppins",
                    "fontSize": 16,
                    "fontWeight": "500",
                  },
                  Object {},
                  Object {
                    "color": "#4D4F4D",
                    "fontFamily": "Poppins",
                    "fontSize": 16,
                    "fontWeight": "500",
                    "lineHeight": 22,
                  },
                ]
              }
              testID="passwordInput"
              value="12345678"
            />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginRight": 5,
                "opacity": 1,
              }
            }
            testID="passwordInputIcon"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={24}
              bbWidth={24}
              fill="none"
              focusable={false}
              height={24}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "auto",
                    "width": "auto",
                  },
                  Object {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width={24}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M3.90744 8.65134C7.53767 2.44955 16.4624 2.44955 20.0926 8.65134C21.3025 10.7182 21.3025 13.2818 20.0926 15.3487C16.4624 21.5504 7.53766 21.5504 3.90744 15.3487C2.69758 13.2818 2.69758 10.7182 3.90744 8.65134Z"
                  fill={4285700459}
                  fillOpacity={0.24}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                    ]
                  }
                />
                <RNSVGPath
                  d="M3.90744 8.65134L3.26018 8.27246L3.26018 8.27246L3.90744 8.65134ZM20.0926 8.65134L20.7399 8.27246V8.27246L20.0926 8.65134ZM20.0926 15.3487L19.4454 14.9698L20.0926 15.3487ZM3.90744 15.3487L4.55471 14.9698L3.90744 15.3487ZM4.55471 9.03021C7.89528 3.32326 16.1048 3.32326 19.4454 9.03021L20.7399 8.27246C16.82 1.57585 7.18005 1.57585 3.26018 8.27246L4.55471 9.03021ZM19.4454 9.03021C20.5183 10.8631 20.5183 13.1369 19.4454 14.9698L20.7399 15.7275C22.0868 13.4266 22.0868 10.5734 20.7399 8.27246L19.4454 9.03021ZM19.4454 14.9698C16.1048 20.6767 7.89528 20.6767 4.55471 14.9698L3.26018 15.7275C7.18005 22.4242 16.82 22.4242 20.7399 15.7275L19.4454 14.9698ZM4.55471 14.9698C3.48182 13.1369 3.48182 10.8631 4.55471 9.03021L3.26018 8.27246C1.91333 10.5734 1.91333 13.4266 3.26018 15.7275L4.55471 14.9698ZM14.8067 12.0607C14.8067 13.6528 13.5387 14.9233 11.9995 14.9233V16.4233C14.3887 16.4233 16.3067 14.4595 16.3067 12.0607H14.8067ZM11.9995 14.9233C10.4605 14.9233 9.19336 13.6531 9.19336 12.0607H7.69336C7.69336 14.4592 9.60993 16.4233 11.9995 16.4233V14.9233ZM9.19336 12.0607C9.19336 10.467 10.4607 9.19699 11.9995 9.19699V7.69699C9.60978 7.69699 7.69336 9.66125 7.69336 12.0607H9.19336ZM11.9995 9.19699C13.5386 9.19699 14.8067 10.4673 14.8067 12.0607H16.3067C16.3067 9.66094 14.3889 7.69699 11.9995 7.69699V9.19699Z"
                  fill={4285700459}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "marginTop": 10,
              "opacity": 1,
            }
          }
        >
          <Text
            adjustsFontSizeToFit={true}
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "textDecorationColor": "#72996B",
                  "textDecorationLine": "underline",
                },
                Object {
                  "color": "#72996B",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "lineHeight": 22,
                },
              ]
            }
          >
            Forgot password ?
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 0.5,
            "justifyContent": "flex-end",
            "marginBottom": 100,
            "marginHorizontal": 20,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#72996B",
              "borderColor": "#72996B",
              "borderRadius": 20,
              "justifyContent": "center",
              "margin": 0,
              "opacity": 1,
              "paddingHorizontal": 15.5,
              "paddingVertical": 24,
            }
          }
          testID="signInButton"
        >
          <Text
            adjustsFontSizeToFit={true}
            style={
              Array [
                Object {
                  "fontFamily": "Poppins",
                },
                Object {
                  "flexGrow": 1,
                  "textAlign": "center",
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": undefined,
                  "fontWeight": "600",
                  "lineHeight": undefined,
                },
              ]
            }
          >
            Sign In
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
